packages/slice-machine/tests/lib/utils/obj.test.ts

```diff
@@ -0,0 +1,56 @@
+import "@testing-library/jest-dom";
+import { deepMerge } from "../../../lib/utils/obj";
+
+describe("deepMerge", () => {
+  test("shallow value - updates value", () => {
+    const obj1 = { val1: 50, val2: 60 };
+    const obj2 = { val2: "hello" };
+    const expected = { val1: 50, val2: "hello" };
+    expect(deepMerge(obj1, obj2)).toStrictEqual(expected);
+    expect(obj1).toStrictEqual({ val1: 50, val2: 60 });
+    expect(obj2).toStrictEqual({ val2: "hello" });
```

Comment:
testing that the original parameters remain unchanged. These assertions would fail if you just use the loadash `merge` function.